### PR TITLE
ISPN-7021 Fix SiteProviderTopologyChangeTest

### PR DESCRIPTION
**Pull request by pruivo**
* fix random failures in SiteProviderTopologyChangeTest

### CODE REVIEW

**Comment by ryanemerson** [comment]
There are two xsite failures, are they related to your `XSiteStateTransferManagerImpl.java ` changes?

File: core/src/test/java/org/infinispan/xsite/statetransfer/failures/SiteProviderTopologyChangeTest.java [hunk 1]
```diff
@@ -238,7 +193,9 @@ private void doXSiteStateTransferDuringTopologyChange(TopologyEvent event) throw
       startStateTransfer(testCaches.coordinator, NYC);
       assertOnline(LON, NYC);
 
+      topologyManager.stopBlocking();
       blockedTopology.unblock();
+      topologyManager.unblockAll();
```
Comment (danberindei) [inline core/src/test/java/org/infinispan/xsite/statetransfer/failures/SiteProviderTopologyChangeTest.java:198]: IMO `unblockAll()` doesn't make sense without `stopBlocking()`, so it should either be controlled by a parameter or the default behaviour of `stopBlocking()`.

Currently `stopBlocking()` logs an error when there are unblocked topologies in the queue, which I think (now) is not a good compromise: it should either throw an exception or unblock all queued topologies (and log only a debug message).

**Review by danberindei — CHANGES_REQUESTED** [review]

File: core/src/test/java/org/infinispan/xsite/statetransfer/failures/SiteProviderTopologyChangeTest.java [hunk 2]
```diff
@@ -35,8 +32,7 @@
  * @author Pedro Ruivo
  * @since 7.0
  */
-//unstable: it looks like not all cases are handled properly. Needs to be revisited! ISPN-6228
-@Test(groups = {"xsite", "unstable"}, testName = "xsite.statetransfer.failures.SiteProviderTopologyChangeTest")
+@Test(groups = "xsite", testName = "xsite.statetransfer.failures.SiteProviderTopologyChangeTest")
```
[the remark this answers is not in the document]
Comment (pruivo) [reply core/src/test/java/org/infinispan/xsite/statetransfer/failures/SiteProviderTopologyChangeTest.java:35]: no. in this PR I'm *trying* to fix the sender side.
ISPN-6228 is about the receiving site.

[the remark this answers is not in the document]
Comment (danberindei) [reply core/src/test/java/org/infinispan/xsite/statetransfer/failures/SiteProviderTopologyChangeTest.java:35]: I was worried that if ISPN-6228 isn't resolved then the random failures will appear again, but if you think ISPN-7021 was the real cause then it's fine. Maybe you should cross that line in the ISPN-6228 description, though ;)

File: core/src/test/java/org/infinispan/xsite/AbstractTwoSitesTest.java [hunk 3]
```diff
@@ -73,7 +73,7 @@ protected ConfigurationBuilder lonConfigurationBuilder() {
             .strategy(lonBackupStrategy)
             .failurePolicyClass(lonCustomFailurePolicyClass)
             .useTwoPhaseCommit(use2Pc)
-            .stateTransfer().maxRetries(1).waitTime(500)
+            .stateTransfer().maxRetries(5).waitTime(1000)
```
[the remark this answers is not in the document]
Comment (pruivo) [reply core/src/test/java/org/infinispan/xsite/AbstractTwoSitesTest.java:76]: the test does not wait for all retries.
one of the causes of random failure was the sender site giving up too soon. The test only waits for the cluster state transfer and cross-site state transfer to complete.

**Comment by pruivo** [comment]
closing this for now. Probably need a new impl for xsite state transfer

**Pull request closed without merging**